src/bun.js/test/expect.zig

```diff
@@ -2807,6 +2935,33 @@ pub const Expect = struct {
         return .zero;
     }
 
+    pub fn toBeValidDate(this: *Expect, globalThis: *JSGlobalObject, callFrame: *CallFrame) callconv(.C) JSValue {
+        defer this.postMatch(globalThis);
+
+        const thisValue = callFrame.this();
+        const value: JSValue = this.getValue(globalThis, thisValue, "toBeValidDate", "") orelse return .zero;
+
+        active_test_expectation_counter.actual += 1;
+
+        const not = this.flags.not;
+        const pass = (value.isDate() and !std.math.isNan(value.getUnixTimestamp())) != not;
```

Comment:
I'm not sure if it's possible, but this would pass for an invalid date if the value is `JSDate` and the internal number is NAN